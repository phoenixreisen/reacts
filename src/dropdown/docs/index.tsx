import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import { Dropdown } from '../dropdown.r';
import React from 'react';

export const Root = () => {
    return (
        <Dropdown title="Awesome Dropdown">
            <a href="#">Link 1</a>
            <a href="#">Link 2</a>
            <a href="#">Link 3</a>
        </Dropdown>
    );
};

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
