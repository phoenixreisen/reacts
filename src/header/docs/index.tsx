import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Header, Topbar } from '../index.r';
import React, { useState } from 'react';
import ReactDOM from 'react-dom';

export const Root = () => {
    const [open, setOpen] = useState<boolean>(false);

    return (<>
        <div className="wrapper wrapper--large">
            <Header
                version="Header mit Topbar Demo"
                toggleNav={(e) => { e.preventDefault(); setOpen(!open) }}
            />
        </div>
        <Topbar
            name="Phoenix Demo App"
            backUrl="https://www.phoenixreisen.com"
            toggleNav={(e) => { e.preventDefault(); setOpen(!open) }}
            toggleAvatar={(e) => { e.preventDefault(); setOpen(!open) }}
        />
    </>);
};

ReactDOM.render(
    <Router>
        <Switch>
            <Route path="/" render={() => <Root />} />
        </Switch>
    </Router>,
    document.querySelector('.example-app'),
);