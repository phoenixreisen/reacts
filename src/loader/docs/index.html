<!DOCTYPE html>
<html lang="de">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <!-- <link rel="stylesheet" type="text/css" href="http://localhost:3027/phx-design-system.css" /> -->
    <link rel="stylesheet" type="text/css" href="https://design-system.phoenixreisen.net/phx-design-system.css" />

    <title>Phoenix Loader Example</title>
</head>
<body>
    <header class="wrapper">
        <div class="example-app"></div>
    </header>

    <script>
        window.process = {
            env: {
                NODE_ENV: 'development'
            }
        }
    </script>
    <script src="./demo.min.js"></script>
</body>
</html>