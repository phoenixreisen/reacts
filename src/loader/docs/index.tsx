import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import React from 'react';
import Loader from '..';

export const Root = () => <Loader text="Demo wird geladen..." />;

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
