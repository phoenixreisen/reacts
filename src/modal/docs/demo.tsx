import { Item1 } from "./demo.content";
import React, { useState } from "react";
import Modal from "../modal.r";

//--- Komponente -----

export const Demo = () => {
  const [show, setShow] = useState(false);

  return (
    <div className="modal-demo">
      <div className="box">
        <a
          href="#"
          title="Modal aufrufen"
          className="btn btn--link"
          onClick={(e) => {
            e.preventDefault();
            setShow(true);
          }}
        >
          Klick mich!
        </a>
      </div>

      {show && (
        <Modal
          title="Modal Demo"
          withCloseText={true}
          toggle={() => setShow(false)}
          preventOverflowHidden
          allowBgClickToClose
          content={
            <div>
              <Item1 />
            </div>
          }
          footer={
            <div className="tr">
              <button
                className="btn btn--secondary mr1"
                onClick={() => setShow(false)}
              >
                Abbrechen
              </button>
              <button
                className="btn btn--primary ml1"
                onClick={() => setShow(false)}
              >
                Verbindlich buchen
              </button>
            </div>
          }
        ></Modal>
      )}
    </div>
  );
};

export default Demo;
