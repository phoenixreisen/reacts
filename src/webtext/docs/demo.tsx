import React from 'react';
import Webtext from '..';

//--- Demo Content -----

export const Webtexts = {
    'webtext1': 'Dies ist der erste Webtext im Webtext-Objekt.',
    'webtext2': 'Dies ist der zweite Webtext im Webtext-Objekt.',
    'webtext3': 'Dies ist der dritte Webtext im Webtext-Objekt.',
    'webtext4': 'Dies ist der vierte Webtext im Webtext-Objekt.',
    'webtext5': 
        `<div>
            <a href="https://www.phoenixreisen.com">Phoenixreisen</a> macht 
            in <span classname="c-primary">Kreufahrten</span>. Ich bekomme aber 
            keine <strong>Suite</strong> mehr. Das 'code'-Element, um <code><strong>dieses 
            Wort</strong></code> sollte gestript worden sein.
        </div>`,
    'webtext6': `
        Hier sind {{anzahl}} Platzhalter enthalten, die ersetzt werden. 
        Die {{eigenschaft}} landen sehr schnell im Gefängnis. {{name}}
    `,
    'webtext7': `
## H2 Markdown Überschrift
### H3 Markdown Überschrift

==Etwas== Text mit *kursiv* und **fett** und einem {{placeholder}}.
Hier noch ein Link zur Website: https://www.phoenixreisen.com/

* Liste
* mit
* Punkten

Hier wieder etwas Text. :-)`,
}

//--- Tooltip Component in several variations -----

export const Demo = () => {
    return (
        <section className="webtext-demo">
            <h3>Webtext mit Namen im title-Attribut</h3>

            <Webtext
                webtexts={Webtexts}
                webtextName="webtext3"
                showWebtextName={true}
                cssClass="webtext-demo-1"
            />

            <h3>Webtext, der seinen Namen nicht preisgibt</h3>

            <Webtext
                webtexts={Webtexts}
                webtextName="webtext4"
                showWebtextName={false}
                cssClass="webtext-demo-2"
            />

            <h3>Webtext, dessen übergebener Name nicht im Objekt existiert</h3>

            <Webtext
                webtexts={Webtexts}
                showWebtextName={true}
                cssClass="webtext-demo-3"
                webtextName="webtext-which-does-not-exist"
            />

            <h3>Webtext mit HTML</h3>

            <Webtext
                webtexts={Webtexts}
                webtextName="webtext5"
                showWebtextName={true}
                cssClass="webtext-demo-4"
            />

            <h3>Webtext mit HTML, aber als Plaintext ausgegeben</h3>

            <Webtext
                asPlainText={true}
                webtexts={Webtexts}
                webtextName="webtext5"
                showWebtextName={false}
                cssClass="webtext-demo-5"
            />

            <h3>Webtext mit Link zum Webtext Manager</h3>

            <Webtext
                webtexts={Webtexts}
                webtextName="webtext3"
                showWebtextName={true}
                wtmLink={'https://meinereise.phoenixreisen.com/cdn/mitarbeiter/webtext-verwaltung/#!form/tefra__text/MeinPhoenix'}
            />

            <h3>HTML-Webtext mit Link zum Webtext Manager</h3>

            <Webtext
                webtexts={Webtexts}
                webtextName="webtext5"
                showWebtextName={true}
                wtmLink={'https://meinereise.phoenixreisen.com/cdn/mitarbeiter/webtext-verwaltung/#!form/tefra__text/MeinPhoenix'}
            />

            <h3>Webtext mit Alternativtext</h3>

            <Webtext
                webtexts={Webtexts}
                showWebtextName={false}
                webtextName="gibts-nicht"
                altText={'Ich werde stattdessen angezeigt.'}
            />

            <Webtext
                webtexts={Webtexts}
                showWebtextName={false}
                webtextName="gibts-nicht"
                altText={'<p>Ich bin <em>HTML</em> als <strong>String</strong>, werder aber gerendert.</p>'}
            />

            <h3>Webtext mit ersetzten Platzhaltern</h3>

            <Webtext
                webtexts={Webtexts}
                showWebtextName={false}
                webtextName="webtext6"
                placeholders={[
                    ['{{anzahl}}', '3'],
                    ['{{name}}', 'Ham Berger'],
                    ['{{eigenschaft}}', 'aggressiven'],
                ]}
            />

            <h3>Webtext mit interpretiertem Markdown</h3>

            <Webtext
                asMarkdown={true}
                webtexts={Webtexts}
                showWebtextName={false}
                webtextName="webtext7"
                placeholders={[
                    ['{{placeholder}}', 'Platzhalter im Markdown'],
                ]}
            />
        </section>
    );
};