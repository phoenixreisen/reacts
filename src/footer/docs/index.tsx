import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { Demo } from './footer.demo';
import ReactDOM from 'react-dom';
import React from 'react';

export const Root = () => {
    return (
        <Demo />
    );
};

ReactDOM.render(
    <Router>
        <Switch>
            <Route path="/" render={() => <Root />} />
        </Switch>
    </Router>,
    document.querySelector('.example-app'),
);
