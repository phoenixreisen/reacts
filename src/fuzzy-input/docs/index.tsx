import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { createRoot } from 'react-dom/client';
import { Demo } from './fuzzy.demo';
import React from 'react';

export const Root = () => {
    return (
        <Demo />
    );
};

createRoot(document.querySelector('.example-app')).render(
    <Router>
        <Routes>
            <Route path="*" element={<Root />} />
        </Routes>
    </Router>
);
