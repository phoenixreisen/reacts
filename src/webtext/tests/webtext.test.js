/**
 * @jest-environment jsdom
 */
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import Webtext, { ALLOWED_HTML } from '../webtext.r';
import { Content, Webtexts } from '../docs/demo'; 
import renderer from 'react-test-renderer';
import striptags from 'striptags';
import React from "react";

describe('Webtext should', () => {
    
    test('have some default classes', () => {
        const component = renderer.create(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                showWebtextName={true}
                cssClass='test-class'
                asPlainText={true}
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('output correct webtext of webtext object', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
            />
        );
        expect(document.querySelector('.webtext').textContent).toContain(Webtexts.webtext3);
    });

    test('show webtext name', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                showWebtextName={true}
            />
        );
        expect(document.querySelector('.webtext').getAttribute('title')).toBe('webtext3');
    });

    test('hide webtext name', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                showWebtextName={false}
            />
        );
        expect(document.querySelector('.webtext').getAttribute('title')).toBeFalsy();
    });

    test('render plain text', () => {
        const component = render(
            <Webtext
                asPlainText={true}
                webtexts={Webtexts}
                webtextName='webtext5'
            />
        );
        expect(document.querySelector('.webtext').textContent).toBe(Webtexts.webtext5);
        expect(document.querySelector('.webtext').innerHTML).toContain('&lt;code&gt;');
    });

    test('render html', () => {
        const component = render(
            <Webtext
                asPlainText={false}
                webtexts={Webtexts}
                webtextName='webtext5'
            />
        );
        const stripped = striptags(Webtexts.webtext5, ALLOWED_HTML);
        expect(document.querySelector('.webtext').innerHTML).toContain('<strong>');
        expect(document.querySelector('.webtext').innerHTML).not.toContain('<code>');
        expect(document.querySelector('.webtext').innerHTML).toBe(`<div>${stripped}</div>`);
    });

    test('insert given css class', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                cssClass='test-class'
            />
        );
        expect(document.querySelector('.webtext').classList.contains('test-class')).toBeTruthy();
    });

    test('add a icon linked to webtext manager as suffix', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                showWebtextName={true}
                wtmLink='https://www.phoenixreisen.com'
                wtmLinkTitle='Phoenix Reisen Website aufrufen'
            />
        );
        expect(document.querySelector('.webtext a')).toBeDefined();
        expect(document.querySelector('.fa-external-link-alt')).toBeDefined();
        expect(document.querySelector('.webtext a').getAttribute('href')).toBe('https://www.phoenixreisen.com');
        expect(document.querySelector('.webtext a').getAttribute('title')).toBe('Phoenix Reisen Website aufrufen');
    });

    test('show webtextName in wtmLink title', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                showWebtextName={true}
                wtmLink='https://www.phoenixreisen.com'
            />
        );
        expect(document.querySelector('.webtext a').getAttribute('title')).toBe('webtext3 im Webtext-Manager Ã¶ffnen');
    });

    test('only show WTM link when showWebtextName is set', () => {
        const component = render(
            <Webtext
                webtexts={Webtexts}
                webtextName='webtext3'
                showWebtextName={false}
                wtmLink='https://www.phoenixreisen.com'
                wtmLinkTitle='Phoenix Reisen Website aufrufen'
            />
        );
        expect(document.querySelector('.webtext a')).toBeNull();
        expect(document.querySelector('.fa-external-link-alt')).toBeNull();
    });
});