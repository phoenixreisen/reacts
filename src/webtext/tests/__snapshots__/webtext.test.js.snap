// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webtext should have some default classes 1`] = `
<article
  className="webtext test-class"
  title="webtext3"
>
  <div>
    Dies ist der dritte Webtext im Webtext-Objekt.
  </div>
</article>
`;
